### DISCUSSION TITLE
using flow to wrap functions sometimes breaks entire app

### DISCUSSION THREAD

**Discussion by 1mike12** [mobxjs/mobx-state-tree#1895] [Q&A] [answered]
For some reason, sometimes when I work with flow, it's bugs out my app in react native. I have some stores that already have flow and they seem to work fine, yet for others when I wrap functions my entire app breaks. But there's no error, just all the images don't load, and there are no responses to normal clicks, gestures etc. I haven't been able to find anyone else with this issue on the internet

I asked about this in an issue back in oct https://github.com/mobxjs/mobx-state-tree/issues/1820

**Comment by EmilTholin** [comment]
Hi @1mike12!

Hmm, very interesting. Do you think you could create a minimal repository that reproduces your problem that we can take a look at?

**Reply by 1mike12** [reply]
Hey @EmilTholin I just finished making the repro project. My project is based on infinite red's https://github.com/infinitered/ignite boilerplate. But the interesting thing is that when created a new project based from scratch, I could not repro the bug. However, taking my current app and whittling it down to the bare parts, I _was_ able to still reproduce it. I really appreciate any help on this. It's such a strange bug. Thanks!

https://github.com/1mike12/flow_example_broken

**Reply by rdewolff** [reply]
How are u able to reproduce the bug? Could you extract the specific `flow()` method breaking the store?

**Reply by 1mike12** [reply]
Yes absolutely. In https://github.com/1mike12/flow_example_broken, there's only one place you can uncoment a flow and you can break it.

**Comment by 1mike12** [comment, marked as answer]
Just in case any poor soul from the future:

So I tried creating a complete barebones app that still had the bug, I tried creating a brand new app from boilerplate which didn't have the bug. I ended up manually copying everyting from the old app under /app and other specific project config files to a new boilerplate. Everything worked afterwards. But even viewing the diff, between right before and after copying, there was nothing that stood out that could explain the problem.

**Reply by rdewolff** [reply]
damn that is scary! 😱
